docs/root/intro/arch_overview/data_sharing_between_filters.rst

```diff
@@ -0,0 +1,93 @@
+.. _arch_overview_data_sharing_between_filters:
```

Comment:
Where do you plan on looking this from? Do we need to add a filter programming section to arch overview? Or some other section? Do you mind taking a look and trying to find the best place to fit this in? Over time I would love to get more of a filter/extension programming guide going so maybe add to here? https://www.envoyproxy.io/docs/envoy/latest/extending/extending. Thoughts?